test/integration/homepage_test.rb

```diff
@@ -0,0 +1,22 @@
+require 'test_helper'
+
+class HomepageTest < ActionDispatch::IntegrationTest
```

Comment:
We talked about IRL.. I reckon it'd be worth using the examples you've created in this PR. The route and controller can stay as it is. Using the example is just as easy and we wouldn't need the condition on the analytics.